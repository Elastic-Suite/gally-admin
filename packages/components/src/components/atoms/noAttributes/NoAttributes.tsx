import React from 'react'
import { styled } from '@mui/system'
import Link from 'next/link'

import Button from '../buttons/Button'
import IonIcon from '../IonIcon/IonIcon'
import { useRouter } from 'next/router'

const CustomRoot = styled('div')(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  gap: '40px',
  padding: theme.spacing(8),
  border: '1px solid',
  borderColor: theme.palette.colors.neutral['300'],
  borderRadius: theme.spacing(1),
  background: theme.palette.colors.white,
  textAlign: 'center',
  alignItems: 'center',
}))

const CustomTitle = styled('div')(({ theme }) => ({
  color: theme.palette.colors.neutral['600'],
  fontSize: '18px',
  lineHeight: '28px',
  fontWeight: '600',
  fontFamily: 'var(--gally-font)',
  width: '550px',
}))

export interface INoAttributesProps {
  title: string
  btnTitle?: string
  btnHref?: string
  absolutLink?: boolean
}

function NoAttributes(props: INoAttributesProps): JSX.Element {
  const { title, btnTitle, btnHref, absolutLink = true } = props

  const { pathname } = useRouter()
  const pathnameWithoutLastFile =
    !absolutLink && pathname
      ? pathname.substring(0, pathname.lastIndexOf('/'))
      : ''

  return (
    <CustomRoot>
      <IonIcon
        name="telescope"
        style={{ color: '#8187B9', width: '40px', height: '40px' }}
      />
      <CustomTitle>{title}</CustomTitle>
      {Boolean(btnTitle) && (
        <Button endIcon={null} startIcon={null} size="large">
          <Link
            href={`${pathnameWithoutLastFile}/${btnHref}`}
            legacyBehavior
            passHref
          >
            <a style={{ textDecoration: 'none' }}>{btnTitle}</a>
          </Link>
        </Button>
      )}
    </CustomRoot>
  )
}

export default NoAttributes
