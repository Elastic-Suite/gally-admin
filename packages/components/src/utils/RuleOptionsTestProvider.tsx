import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
