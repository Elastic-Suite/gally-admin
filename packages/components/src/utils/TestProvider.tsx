import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
