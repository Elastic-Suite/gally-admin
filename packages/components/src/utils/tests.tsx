import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
