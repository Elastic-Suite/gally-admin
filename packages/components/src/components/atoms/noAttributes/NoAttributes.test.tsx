import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import NoAttributes from './NoAttributes'

describe('NoAttributes', () => {
  it('NoAttributes with absolutLink', () => {
    const { container } = renderWithProviders(
      <NoAttributes
        title="No attributes were specified as searchable in the settings"
        btnTitle="Add searchable attributes"
        btnHref="admin/settings/attributes"
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('NoAttributes without absolutLink', () => {
    const { container } = renderWithProviders(
      <NoAttributes
        title="No attributes were specified as searchable in the settings"
        btnTitle="Add searchable attributes"
        btnHref="admin/settings/attributes"
        absolutLink={false}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
