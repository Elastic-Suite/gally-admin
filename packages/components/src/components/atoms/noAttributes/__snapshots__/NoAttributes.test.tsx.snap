// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoAttributes NoAttributes with absolutLink 1`] = `
<div>
  <div
    class="css-1tomd0t"
  >
    <ion-icon
      data-testid="ionicon"
      src="/images/ionicons/custom/telescope-outline.svg"
      style="color: rgb(129, 135, 185); width: 40px; height: 40px;"
    />
    <div
      class="css-pbnn4f"
    >
      No attributes were specified as searchable in the settings
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-zwkw2d-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      tabindex="0"
      type="button"
    >
      <a
        class="css-honi22"
        href="/admin/settings/attributes"
      >
        Add searchable attributes
      </a>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`NoAttributes NoAttributes without absolutLink 1`] = `
<div>
  <div
    class="css-1tomd0t"
  >
    <ion-icon
      data-testid="ionicon"
      src="/images/ionicons/custom/telescope-outline.svg"
      style="color: rgb(129, 135, 185); width: 40px; height: 40px;"
    />
    <div
      class="css-pbnn4f"
    >
      No attributes were specified as searchable in the settings
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-zwkw2d-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      tabindex="0"
      type="button"
    >
      <a
        class="css-honi22"
        href="/admin/settings/attributes"
      >
        Add searchable attributes
      </a>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
